src/include/network/connection_context.h

```diff
@@ -13,6 +13,21 @@ namespace terrier::network {
  */
 
 struct ConnectionContext {
+  /**
+   * Commandline arguments parsed from protocol interpreter
+   */
+  std::unordered_map<std::string, std::string> cmdline_args_;
+
+  /**
+   * The OID of the database accessed by this connection
+   */
+  catalog::db_oid_t db_oid_ = catalog::INVALID_DATABASE_OID;
```

Comment:
Since it's just a struct we don't have much control over how people use this. If we don't want to make it a proper class, we should document some invariants about db_oid_. Presumably it shouldn't change after the connection is established, and then can't be changed again until the connection is closed and then the context can possibly be reused?